### PR TITLE
Add synchronous redraw API to AnimationLoop

### PR DESCRIPTION
**Pull request by Pessimistress**
Per discussion

### CODE REVIEW

File: modules/core/src/core/animation-loop.js [hunk 1]
```diff
@@ -205,20 +225,11 @@ export default class AnimationLoop {
...
-    this.onRender(this.animationProps);
-    // end callback
-
-    // clear needsRedraw flag
-    this._clearNeedsRedraw();
+    const promise = this.redraw();
 
-    if (this.offScreen && this.gl.commit) {
-      // https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/commit
-      // commit returns a Promise
```
Comment (ibgreen) [inline modules/core/src/core/animation-loop.js]: According to the linked doc, `commit` is a void function. Maybe this has changed as the offscreen feature evolved?

File: modules/core/src/core/animation-loop.js [hunk 2]
```diff
@@ -205,20 +225,11 @@ export default class AnimationLoop {
...
-      this.gl.commit().then(this._renderFrame);
+    if (promise) {
+      // offscreen rendering
+      promise.then(this._renderFrame);
```
Comment (ibgreen) [inline modules/core/src/core/animation-loop.js]: This looks recursive? Did you mean to call a different function?

File: modules/core/src/core/animation-loop.js [hunk 3]
```diff
@@ -136,6 +136,26 @@ export default class AnimationLoop {
...
+
+    // clear needsRedraw flag
+    this._clearNeedsRedraw();
+
+    if (this.offScreen && this.gl.commit) {
+      // https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/commit
+      // commit returns a Promise
+      return this.gl.commit();
+    }
+    return null;
```
Comment (ibgreen) [inline modules/core/src/core/animation-loop.js]: If the return value is no longer a promise (see below), maybe just return this?

File: modules/core/src/core/animation-loop.js (hunk 1 above)
Comment (Pessimistress) [reply modules/core/src/core/animation-loop.js]: Good catch. I haven't checked back on the doc for a while.

**Pull request merged**